modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/utils.py

```diff
@@ -78,3 +82,59 @@ def _categories(self):  # noqa: GL08
     def _set_categories(self, categories):  # noqa: GL08
         self._lazy_categories = categories
         self._table = None
+
+
+def check_join_supported(join_type: str):
+    """
+    Check if join type is supported by HDK.
+
+    Parameters
+    ----------
+    join_type : str
+        Join type.
+
+    Returns
+    -------
+    None
+    """
+    if join_type not in ("inner", "left"):
+        raise NotImplementedError(
+            f"{join_type} join is not supported by the HDK engine"
+        )
+
+
+def arrow_to_pandas(at: pa.Table) -> pd.DataFrame:
+    """
+    Convert the specified arrow table to pandas.
+
+    Parameters
+    ----------
+    at : pyarrow.Table
+        The table to convert.
+
+    Returns
+    -------
+    pandas.DataFrame
+    """
+    return at.to_pandas(types_mapper=_arrow_to_pandas_mapper)
+
+
+def _arrow_to_pandas_mapper(at):  # noqa: GL08
+    if is_dictionary(at) and isinstance(at.value_type, ArrowIntervalType):
+        # The default mapper fails with TypeError: unhashable type: 'dict'
+        return _CategoricalDtypeMapper
+    return None
```

Comment:
Maybe so? Since the function is small and is not used anywhere else.

```suggestion
    def _arrow_to_pandas_mapper(at):
        if is_dictionary(at) and isinstance(at.value_type, ArrowIntervalType):
            # The default mapper fails with TypeError: unhashable type: 'dict'
            return _CategoricalDtypeMapper
        return None

    return at.to_pandas(types_mapper=_arrow_to_pandas_mapper)
```